src/device/selection.ts

```diff
@@ -140,19 +147,55 @@ export const offsetStart = (): number | null => {
   return range.startOffset || 0
 }
 
+/** Recursively find for saved node. */
+const findNodeRecusivelyFromRoot = (
+  root: Element | ChildNode | Node | null | undefined,
+  savedFocusNode: Node,
+  savedFocusOffset: number,
+): ChildNode | undefined => {
+  if (!root) return undefined
+
+  const childNodes = Array.from(root.childNodes)
+  const hasTextNode = childNodes.some(node => node.nodeType === Node.TEXT_NODE)
+  if (hasTextNode) {
+    const focusNode = childNodes.find(node => node.isSameNode(savedFocusNode))
+    return focusNode
```

Comment:
`childNodes.find(node => node.isSameNode(savedFocusNode))` will always return `savedFocusNode | undefined`, since you're just finding the child that is `savedFocusNode`. From what I can tell, `findNodeRecursivelyRoot` traverses from a root node to a given node, just to return the given node itself. Am I missing something?

Reply:
Yes, you're right. I added it since it was not working at that moment, and I was using `sel.addRange` instead of `collapse` then. Later, I changed to `collapse` but forgot to check if `savedFocusNode` directly was working or not. Thanks for this.

Reply:
I checked and right now it works directly when using savedFocusNode. Earlier the added HTML and stripped HTML changed the nodes themselves, and I thought it was necessary, but it seems not.

Reply:
Thanks for the explanation, and for fixing that up. 

Note: As soon as you typed `isSameNode` you should have noticed something was wrong. It should have raised the question, "why am I traversing the DOM just to return something I already have?" I could tell that it was not doing anything just by looking at the code because it was an invariant. No experimental evidence needed. 

The best advice I can give is to pay more attention to what variables are in scope and think about how you might use them without having to do a lot of extra work. This case was obvious, but another more common case is that the function may be necessary, but there is a more efficient/elegant solution. I know that seeing these improvements come with time and experience, so keep working to learn and grow and I'm sure you'll get a knack for it :). 

Lastly, try to get in the mindset that **there is always one best way**, not multiple ways to do the same thing. Looking for the best approach will help guide your mind towards recognizing inefficiencies or clunky implementations.

Reply:
Thanks for the guidance. I will pay more attention to the variables in scope, and focus on the best way to solve them.